e2e/mock-api/src/app/routes.auth.js

```diff
@@ -442,33 +442,41 @@ export default function (app) {
         req.query['authorize-middleware'] === 'authorization' &&
         !req.query['logout-middleware']
       ) {
-        res.redirect(url);
+        res.redirect(loginUrl);
       } else {
         res.status(406).send('Middleware additions are missing.');
       }
     } else {
       if (req.cookies.iPlanetDirectoryPro) {
-        const url = new URL(`${req.query.redirect_uri}`);
-        url.searchParams.set('client_id', 'bar');
-        url.searchParams.set('code', 'foo');
-        url.searchParams.set('iss', `${AM_URL}/oauth2`);
-        url.searchParams.set('state', req.query.state);
-        res.redirect(url);
-      } else if (req.headers.accept.includes('html')) {
-        // we set auth because we are mimicking the html document from central login
-        // this helps our test know it is ready to evaluate the page
-        // and not to evaluate when we have the 401 below
-        url.searchParams.set('auth', true);
-        res.redirect(url);
+        const redirectUrl = new URL(`${req.query.redirect_uri}`);
+
+        console.log(`Request URL: ${req.query.client_id}`);
+
+        redirectUrl.searchParams.set('client_id', req.query.client_id);
+        redirectUrl.searchParams.set('code', 'foo');
+        redirectUrl.searchParams.set('iss', `${AM_URL}/oauth2`);
+        redirectUrl.searchParams.set('state', req.query.state);
+
+        res.redirect(redirectUrl);
+      } else if (req.cookies.redirected === 'true') {
+        res.redirect(loginUrl);
```

Comment:
Improving the redirection in the mock server to help test the full redirects needed for centralized login.